test/e2e/network_segmentation_policy.go

```diff
@@ -0,0 +1,344 @@
+			framework.ExpectNoError(err, "Failed to remove the namespace %s %v", f.Namespace.Name, err)
+			if defaultNetNamespace != "" {
+				err = cs.CoreV1().Namespaces().Delete(context.TODO(), defaultNetNamespace, metav1.DeleteOptions{})
+				framework.ExpectNoError(err, "Failed to remove the namespace %v", defaultNetNamespace, err)
+			}
+
+		}
+
+		ginkgo.AfterEach(func() {
+			cleanupFn()
+		})
+
+		ginkgo.DescribeTable(
+			"pods within namespace should be isolated when deny policy is present",
+			func(
+				netConfigParams networkAttachmentConfigParams,
+				clientPodConfig podConfiguration,
+				serverPodConfig podConfiguration,
+			) {
+				ginkgo.By("Creating the attachment configuration")
+				netConfig := newNetworkAttachmentConfig(netConfigParams)
+				netConfig.namespace = f.Namespace.Name
+				_, err := nadClient.NetworkAttachmentDefinitions(f.Namespace.Name).Create(
+					context.Background(),
+					generateNAD(netConfig),
+					metav1.CreateOptions{},
+				)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				ginkgo.By("creating client/server pods")
+				serverPodConfig.namespace = f.Namespace.Name
+				clientPodConfig.namespace = f.Namespace.Name
+				runUDNPod(cs, f.Namespace.Name, serverPodConfig, nil)
+				runUDNPod(cs, f.Namespace.Name, clientPodConfig, nil)
+
+				var serverIP string
+				for i, cidr := range strings.Split(netConfig.cidr, ",") {
+					if cidr != "" {
+						ginkgo.By("asserting the server pod has an IP from the configured range")
+						serverIP, err = podIPsForUserDefinedPrimaryNetwork(
+							cs,
+							f.Namespace.Name,
+							serverPodConfig.name,
+							namespacedName(f.Namespace.Name, netConfig.name),
+							i,
+						)
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+						const netPrefixLengthPerNode = 24
+						ginkgo.By(fmt.Sprintf("asserting the server pod IP %v is from the configured range %v/%v", serverIP, cidr, netPrefixLengthPerNode))
+						subnet, err := getNetCIDRSubnet(cidr)
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+						gomega.Expect(inRange(subnet, serverIP)).To(gomega.Succeed())
+					}
+
+					ginkgo.By("asserting the *client* pod can contact the server pod exposed endpoint")
+					gomega.Eventually(func() error {
+						return reachToServerPodFromClient(cs, serverPodConfig, clientPodConfig, serverIP, port)
+					}, 2*time.Minute, 6*time.Second).Should(gomega.Succeed())
+				}
+
+				ginkgo.By("creating a \"default deny\" network policy")
+				_, err = makeDenyAllPolicy(f, f.Namespace.Name, "deny-all")
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				ginkgo.By("asserting the *client* pod can not contact the server pod exposed endpoint")
+				gomega.Eventually(func() error {
+					return reachToServerPodFromClient(cs, serverPodConfig, clientPodConfig, serverIP, port)
+				}, 1*time.Minute, 6*time.Second).ShouldNot(gomega.Succeed())
+
+			},
+			ginkgo.Entry(
+				"in L2 dualstack primary UDN",
```

Comment:
the title of this PR says netpol for L3, but here are L2 test cases and changes to layer 2 controller. So is L2 also supported and the title just needs to be updated?

Reply:
yes @trozet , updated title and description accordingly now.